pkg/filter/filter.go

```diff
@@ -0,0 +1,55 @@
+package filter
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/label"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capiutil "sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-admission-controller/pkg/release"
+)
+
+func IsObjectReconciledByLegacyRelease(ctx context.Context, ctrlClient client.Client, ctrlCache client.Reader, objectMeta metav1.Object) (bool, error) {
```

Comment:
@tuommaki  I added both `client.Client` and `client.Reader` here since CAPI `GetClusterByName(...)` func needs `client.Client` and we need `client.Reader` for cached releases.

Not sure if there is some better way to handle this.

Reply:
Wait, I'll just not use CAPI helper func, since it's a regular Get call inside, that will make the func signature prettier.

Reply:
What if this was split into two functions: `FindClusterRelease()` and `IsLegacyRelease()` ?

Reply:
We can, just not sure if there is a need, as this is the CAPI helper func

```
// GetClusterByName finds and return a Cluster object using the specified params.
func GetClusterByName(ctx context.Context, c client.Client, namespace, name string) (*clusterv1.Cluster, error) {
	cluster := &clusterv1.Cluster{}
	key := client.ObjectKey{
		Namespace: namespace,
		Name:      name,
	}

	if err := c.Get(ctx, key, cluster); err != nil {
		return nil, err
	}

	return cluster, nil
}
```

It's literally just a Get call :smile:

Reply:
I can split the func here and make the consumer call 2 funcs, but I think it would be more clear to have one func that answers the question `Should I validate/mutate this or just ignore it?`